src/Symfony/Component/VarDumper/Caster/RdKafkaCaster.php

```diff
@@ -0,0 +1,167 @@
+        ];
+
+        $a += self::extractMetadata($c);
+
+        return $a;
+    }
+
+    public static function castTopic(Topic $c, array $a, Stub $stub, $isNested)
+    {
+        $prefix = Caster::PREFIX_VIRTUAL;
+
+        $a += [
+            $prefix.'name' => $c->getName(),
+        ];
+
+        return $a;
+    }
+
+    public static function castTopicPartition(TopicPartition $c, array $a)
+    {
+        $prefix = Caster::PREFIX_VIRTUAL;
+
+        $a += [
+            $prefix.'offset' => $c->getOffset(),
+            $prefix.'partition' => $c->getPartition(),
+            $prefix.'topic' => $c->getTopic(),
+        ];
+
+        return $a;
+    }
+
+    public static function castMessage(Message $c, array $a, Stub $stub, $isNested)
+    {
+        $prefix = Caster::PREFIX_VIRTUAL;
+
+        $a += [
+            $prefix.'errstr' => $c->errstr(),
+        ];
+
+        return $a;
+    }
+
+    public static function castConf(Conf $c, array $a, Stub $stub, $isNested)
+    {
+        $prefix = Caster::PREFIX_VIRTUAL;
+
+        foreach ($c->dump() as $key => $value) {
+            $a[$prefix.$key] = $value;
+        }
+
+        return $a;
+    }
+
+    public static function castTopicConf(TopicConf $c, array $a, Stub $stub, $isNested)
+    {
+        $prefix = Caster::PREFIX_VIRTUAL;
+
+        foreach ($c->dump() as $key => $value) {
+            $a[$prefix.$key] = $value;
+        }
+
+        return $a;
+    }
+
+    public static function castRdKafka(\RdKafka $c, array $a, Stub $stub, $isNested)
+    {
+        $prefix = Caster::PREFIX_VIRTUAL;
+
+        $a += [
+            $prefix.'out_q_len' => $c->getOutQLen(),
+        ];
+
+        $a += self::extractMetadata($c);
+
+        return $a;
+    }
+
+    private static function prepareAssignment(iterable $partitions)
+    {
+        return array_map(function (TopicPartition $topicPartition) {
+            return self::castTopicPartition($topicPartition, []);
+        }, $partitions);
+    }
+
+    private static function extractMetadata($c)
+    {
+        try {
+            $m = $c->getMetadata(true, null, 500);
+        } catch (RdKafkaException $e) {
+            $m = null;
+        }
+
+        if (!$m) {
+            return [];
+        }
+
+        return [
+            'orig_broker_id' => $m->getOrigBrokerId(),
+            'orig_broker_name' => $m->getOrigBrokerName(),
+            'brokers' => array_map(function (BrokerMetadata $broker) {
+                return [
+                    'id' => $broker->getId(),
+                    'host' => $broker->getHost(),
+                    'port' => $broker->getPort(),
+                ];
+            }, iterator_to_array($m->getBrokers())),
+            'topics' => array_map(function (TopicMetadata $topic) {
+                return [
+                    'name' => $topic->getTopic(),
+                    'partitions' => array_map(function (PartitionMetadata $partition) {
```

Comment:
shouldn't the caster for Topic list partitions?
and shouldn't we let Partition as objects?